packages/moonstone/DropDown/DropDown.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * Moonstone styled DropDown components
+ *
+ * @example
+ * <DropDown
+ * 		defaultSelected={2}
+ *		inline
+ *		title="Dropdown"
+ * >
+ *   {['Option 1', 'Option 2', 'Option 3', 'Option 4', 'Option 5']}
+ * </DropDown>
+ *
+ * @module moonstone/DropDown
+ * @exports DropDown
+ * @exports DropDownBase
+ */
+
+
+import Changeable from '@enact/ui/Changeable';
+import Toggleable from '@enact/ui/Toggleable';
+import equals from 'ramda/src/equals';
+import Group from '@enact/ui/Group';
+import kind from '@enact/core/kind';
+import Pure from '@enact/ui/internal/Pure';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import React from 'react';
+
+import {Button} from '../Button';
+import ContextualPopupDecorator from '../ContextualPopupDecorator';
+import {Item} from '../Item';
+import {Scroller} from '../Scroller';
+
+import css from './DropDown.module.less';
+
+
+const compareChildren = (a, b) => {
+	if (!a || !b || a.length !== b.length) return false;
+
+	let type = null;
+	for (let i = 0; i < a.length; i++) {
+		type = type || typeof a[i];
+		if (type === 'string') {
+			if (a[i] !== b[i]) {
+				return false;
+			}
+		} else if (!equals(a[i], b[i])) {
+			return false;
+		}
+	}
+
+	return true;
+};
+
+const DropDownButton = kind({
+	name: 'DropDownButton',
+
+	styles: {
+		css,
+		className: 'button'
+	},
+
+	render: (props) => (
+		<Button
+			{...props}
+			iconPosition="after"
+			small
+		/>
+	)
+});
+
+const ContextualButton = ContextualPopupDecorator({noArrow: true}, DropDownButton);
+
+const DropDownList = kind({
+	name: 'DropDownList',
+
+	propTypes: /** @lends moonstone/DropDown.DropDownBase.prototype */ {
+		/**
+		 * The selections for Dropdown
+		 *
+		 * @type {String[]}
+		 * @public
+		 */
+		children: PropTypes.node,
+
+		/**
+		 * Called when an item is selected.
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onSelect: PropTypes.func,
+
+		/**
+		 * Index of the selected item.
+		 *
+		 * @type {Number}
+		 * @public
+		 */
+		selected: PropTypes.number
+	},
+
+	styles: {
+		className: 'dropDownList',
+		css
+	},
+
+	render: ({children, onSelect, selected, ...rest}) => {
+		return (
+			<div {...rest}>
+				<Group
+					childComponent={Item}
+					className={css.group}
+					component={Scroller}
```

Comment:
Is anything gained using `Scroller`?  When I run the sample, there isn't really any scrolling to do.  If I resize the sample window small enough and re-run it, the items will render past the sample viewport and you can't scroll them.

Is the expanded dropdown area supposed to be a fixed size or is it supposed to grow depending on its children?

Reply:
The dropdown height depends on its items up to 5; thus the 5 item limit max. So any more than 5 items, there'll be a Scroller.

Reply:
Should we consider having this be conditional to not add a `Scroller` (and its cost) if the list length is 5 or less?